### PR TITLE
Add shortcut property to send modals via ApplicationContext

### PR DESCRIPTION
**Pull request by krittick**
## Summary

This adds a `send_modal` shortcut property to `ApplicationContext` to allow easier sending of modal dialogs instead of needing to use `ctx.interaction.response.send_modal()`

Examples were also updated to use the new method where appropriate.

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

File: discord/commands/context.py [hunk 1]
```diff
@@ -215,10 +215,15 @@ def unselected_options(self) -> Optional[List[Option]]:
                 return self.command.options  # type: ignore
         return None
 
+    @property
+    def send_modal(self) -> None:
```
Comment (plun1331) [inline discord/commands/context.py:219]: Why is this typed to `None`

Comment (krittick) [reply discord/commands/context.py:219]: Because I forgot this branch doesn't have that update to the underlying method. Will remove.

**Comment by Lulalaby** [comment]
Send modal sounds inappropriate
Maybe respond with modal?

**Comment by krittick** [comment]
> Send modal sounds inappropriate
> Maybe respond with modal?

I just kept it the same as the underlying method so it's easier for users to switch.

**Review by Dorukyum — APPROVED** [review]

File: discord/interactions.py [hunk 2]
```diff
@@ -802,6 +802,7 @@ async def send_modal(self, modal: Modal):
         )
         self._responded = True
         self._parent._state.store_modal(modal, self._parent.user.id)
+        return None
```
Comment (Middledot) [inline discord/interactions.py:805]: I don't think this is needed

**Comment by krittick** [comment]
PR has been updated to make `send_modal` now return the `Interaction` object, which mirrors the behavior of `send_message` for consistency.

**Review by Middledot — APPROVED** [review]

**Review by Dorukyum — APPROVED** [review]

**Pull request merged**